[vulkan] jni build support USE_VULKAN

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#39188 [vulkan] jni build support USE_VULKAN**

Extracting Vulkan_LIBS and Vulkan_INCLUDES setup from `cmake/Dependencies.cmake` to `cmake/VulkanDependencies.cmake` and reuse it in android/pytorch_android/CMakeLists.txt

Adding control to build with Vulkan setting env variable `USE_VULKAN` for `scripts/build_android.sh` `scripts/build_pytorch_android.sh`

We do not use Vulkan backend in pytorch_android, but with this build option we can track android aar change with `USE_VULKAN` added.

Currently it is 88Kb.

Differential Revision: [D21770892](https://our.internmc.facebook.com/intern/diff/D21770892)

IvanKobzarev committed:
[vulkan] jni build support USE_VULKAN

[ghstack-poisoned]

dreiss commented on android/pytorch_android/CMakeLists.txt:
```diff
@@ -120,5 +136,10 @@ else()
       cpuinfo
       clog
   )
+  if(USE_VULKAN)
+    list(APPEND pytorch_jni_LIBS ${Vulkan_LIBS})
+  endif()
+  # Prefer dynamic linking on the host
+  target_link_libraries(pytorch_jni ${pytorch_jni_LIBS})
```
Could pull this out of the if/else block if you want.

dreiss commented on cmake/VulkanDependencies.cmake:
```diff
@@ -0,0 +1,163 @@
+  endif(USE_VULKAN_SHADERC_RUNTIME)
+endif()
+
+#set(Vulkan_LIBS ${Vulkan_LIBS} PARENT_SCOPE)
+#set(Vulkan_INCLUDES ${Vulkan_INCLUDES} PARENT_SCOPE)
+
```
Remove?

IvanKobzarev committed:
apply review comments: cleanup, extract common block from if-else on "[vulkan] jni build support USE_VULKAN"


Extracting Vulkan_LIBS and Vulkan_INCLUDES setup from `cmake/Dependencies.cmake` to `cmake/VulkanDependencies.cmake` and reuse it in android/pytorch_android/CMakeLists.txt

Adding control to build with Vulkan setting env variable `USE_VULKAN` for `scripts/build_android.sh` `scripts/build_pytorch_android.sh`

We do not use Vulkan backend in pytorch_android, but with this build option we can track android aar change with `USE_VULKAN` added.

Currently it is 88Kb.

[ghstack-poisoned]
